lib/elixir/lib/version.ex

```diff
@@ -119,6 +119,27 @@ defmodule Version do
     defstruct [:source, :matchspec, :compiled]
 
     @opaque t :: %__MODULE__{source: String.t(), matchspec: :ets.match_spec(), compiled: boolean}
+
+    @doc false
+    def new(source, spec) do
+      %__MODULE__{source: source, matchspec: spec, compiled: false}
+    end
+
+    @doc false
+    def compile(%__MODULE__{matchspec: spec} = req) do
+      %{req | matchspec: :ets.match_spec_compile(spec), compiled: true}
+    end
+
+    @doc false
+    def match?(%__MODULE__{matchspec: spec, compiled: true}, matchable_pattern) do
+      matches = :ets.match_spec_run([matchable_pattern], spec)
+      Enum.any?(matches)
```

Comment:
```suggestion
      matches != []
```

Since the result can only be lists, I prefer a `!= []` instead of Enum which can be polymorphic. :)

Reply:
:+1:

I did this because of how it sounds "Is there any matches?" instead of "is not an empty list". But I can see the benefits of sticking to simple